### PR TITLE
Feature/open menu on create

### PR DESCRIPTION
**Pull request by JedrekDabrowski**

### CODE REVIEW

**Comment by JedrekDabrowski** [comment]
https://github.com/graphql-editor/graphql-editor/issues/502

File: packages/editor/src/Graf/Graf.tsx [hunk 1]
```diff
@@ -169,13 +172,19 @@ export const Graf: React.FC = () => {
...
+                  JSON.stringify(
+                    createParserField({
+                      ...rest,
+                      name: nodeToDuplicate?.name + 'Copy',
+                    }),
+                  ),
                 ) as ParserField;
                 allNodes.push(duplicatedNode);
+                set({
+                  n: duplicatedNode?.id,
```
Comment (aexol) [inline packages/editor/src/Graf/Graf.tsx:186]: redundant. There is an effect on selected node which sets the router state

File: packages/editor/src/shared/components/ContextMenu/ExtendNodeMenu/index.tsx [hunk 2]
```diff
@@ -48,24 +50,25 @@ export const ExtendNodeMenu: React.FC<ExtendNodeMenuProps> = ({ hideMenu }) => {
...
-    );
+      },
+      name: f.name,
+      args: [],
+      interfaces: [],
+      directives: [],
+    });
+    tree.nodes.push(extendNode);
+    setSelectedNode({ field: extendNode, source: 'diagram' });
+    set({ n: extendNode.id });
```
Comment (aexol) [inline packages/editor/src/shared/components/ContextMenu/ExtendNodeMenu/index.tsx:71]: redundant. setSelected node fires an effect in Editor.tsx

File: packages/editor/src/shared/components/PaintNodes/NewNode.tsx [hunk 3]
```diff
@@ -103,12 +104,17 @@ export const NewNode: React.FC<NewNodeProps> = ({ node, onCreate }) => {
...
     useTreesState();
+  const { set } = useRouter();
+
   const isError =
     tree.nodes.map((n) => n.name).includes(newName) ||
     libraryTree.nodes.map((n) => n.name).includes(newName);
   const submit = () => {
     if (newName && !isError) {
       const n = onCreate(newName);
+      set({
```
Comment (aexol) [inline packages/editor/src/shared/components/PaintNodes/NewNode.tsx:115]: redundant

**Review by aexol — CHANGES_REQUESTED** [review]
```ts
  useEffect(() => {
    if (tree.initial || selectedNode?.source === 'routing') {
      return;
    }
    set({
      n: selectedNode?.field?.id,
    });
  }, [selectedNode]);
```
This effect is responsible for using `set` function
This code is in Editor.tsx

**Pull request merged**